tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -1926,3 +1927,158 @@ def call(self, inputs, state):
     new_state = core_rnn_cell.LSTMStateTuple(new_c, new_h)
 
     return new_h, new_state
+
+class GLSTMCell(core_rnn_cell.RNNCell):
+  """Group LSTM cell (G-LSTM).
+
+  The implementation is based on:
+
+    https://arxiv.org/abs/1703.10722
+
+  O. Kuchaiev and B. Ginsburg
+  "Factorization Tricks for LSTM Networks", ICLR 2017 workshop.
+  """
+
+  def __init__(self, num_units, initializer=None, num_proj=None,
+               number_of_groups=1, forget_bias=1.0, activation=tanh):
+    """Initialize the parameters of G-LSTM cell.
+
+    Args:
+      num_units: int, The number of units in the G-LSTM cell
+      initializer: (optional) The initializer to use for the weight and
+        projection matrices.
+      num_proj: (optional) int, The output dimensionality for the projection
+        matrices.  If None, no projection is performed.
+      number_of_groups: (optional) int, number of groups to use. If number_of_groups=1,
+        then it should be equivalent to LSTMP cell
+      forget_bias: Biases of the forget gate are initialized by default to 1
+        in order to reduce the scale of forgetting at the beginning of
+        the training.
+      activation: Activation function of the inner states.
+    """
+    self._num_units = num_units
+    self._initializer = initializer
+    self._num_proj = num_proj
+    self._forget_bias = forget_bias
+    self._activation = activation
+    self._number_of_groups = number_of_groups
+
+    assert (self._num_units % self._number_of_groups == 0)
+    if self._num_proj:
+      assert (self._num_proj % self._number_of_groups == 0)
+      self._group_shape = [int(self._num_proj / self._number_of_groups),
+                           int(self._num_units / self._number_of_groups)]
+    else:
+      self._group_shape = [int(self._num_units / self._number_of_groups),
+                           int(self._num_units / self._number_of_groups)]
+
+    if num_proj:
+      self._state_size = (core_rnn_cell.LSTMStateTuple(num_units, num_proj))
+      self._output_size = num_proj
+    else:
+      self._state_size = (core_rnn_cell.LSTMStateTuple(num_units, num_units))
+      self._output_size = num_units
+
+  @property
+  def state_size(self):
+    return self._state_size
+
+  @property
+  def output_size(self):
+    return self._output_size
+
+  def _get_input_for_group(self, inpt, group_id, group_size):
+    """Slices inputs into groups to prepare for processing by cell's groups
+
+    Args:
+      inpt: inputs
+      group_id: group id, for which to prepare extract input_group_id
+      group_size: size of the group
+
+    Returns:
+      subset of inputs corresponding to group "group_id"
+    """
+    return array_ops.slice(input_=inpt,
+                           begin=[0, group_id * group_size],
+                           size=[inpt.get_shape()[0].value, group_size],
+                           name="GLSTMinputGroupCreation")
+
+  def __call__(self, inputs, state):
```

Comment:
def call(self, inputs, state):

RNNCell provides `__call__` for you as of last week.